src/Lachain.Core/Blockchain/SystemContracts/NativeTokenContract.cs

```diff
@@ -128,6 +147,48 @@ public ExecutionStatus Approve(UInt160 spender, UInt256 amount, SystemContractEx
             frame.ReturnValue = 1.ToUInt256().ToBytes();
             return ExecutionStatus.Ok;
         }
+        
+        [ContractMethod(Lrc20Interface.MethodSetAllowedSupply)]
+        public ExecutionStatus SetAllowedSupply(Money amount, SystemContractExecutionFrame frame)
+        {
+            frame.UseGas(GasMetering.NativeTokenApproveCost);
+            if(!frame.InvocationContext.Sender.Equals(_mintCntrlAdd))
+                return ExecutionStatus.ExecutionHalted;
+            
+            if(_allowedSupply + amount > _maxSupply)
```

Comment:
We should check just amount:
```suggestion
            if(amount > _maxSupply)
```

Reply:
incorporated